translated: docs/developer-docs/integrations/internet-identity

hokosugi opened the pull request:
## 疑問点・不明点

integrate-identity.md
* 30行目 `hence "caller"` をコーラーとしましたが、「呼び出し人」か原文で caller でも良い気がしています。
* 32行目 `IC adds to requests` がイマイチ理解出来ていません。「ICがリクエストに追加する」と直訳しましたが、「ICがリクエストに答える」くらいの意味なのかも知れません。
* 74行目 `read: non-documented` が全く分かりません。原文のままにしました。筆者の備忘録で消し忘れたのではないかと思いますが、どうなんでしょうか？

alternative-origins.md
* タイトル 「代替フロントエンドオリジン」としましたが、「代替オリジン」なら良いのですが、フロントエンドを挟むと原文のままのほうが(Alternative Frontend Origins)のほうがいいのかもと思ったりします。
* 9~12行目 動名詞を「〜する」としましたが、名詞で切るほうが良かったでしょうか？
* 76行目 （隠しファイル）を挿入しましたが、要らないときは消してください。
* 107行目 `content-type` を原文のままにしました。「コンテンツタイプヘッダー」というサイトもありますが、原文のままほうが多い印象です。

なお、当初1ファイルでしたが、本家とマージした結果2ファイルになっています。

では宜しくおねがいします。

hokosugi committed:
traslated: docs/developer-docs/integrations/internet-identity

hoosan commented on docs/developer-docs/integrations/internet-identity/integrate-identity.md:
```diff
@@ -1,3 +1,113 @@
+#  Internet Identity の統合
+
+これは、Internet Identity を使用してプロジェクトを統合し、テストする方法を示しています。Internet Identity の開発版 [build flavor](https://github.com/dfinity/internet-identity/blob/main/README.md#build-features-and-flavors) と [agent-js](https://github.com/dfinity/agent-js) ライブラリを使用します。
```
「ここでは」とするのはどうでしょうか。
また、開発／プロダクション版などの切り替えであることを示すフレーバーは一般用語の範疇としてカタカナ&
英語注記でも良いと思います。
```suggestion
ここでは、Internet Identity を使用してプロジェクトを統合し、テストする方法を示します。Internet Identity の開発版 [ビルドフレーバー（Flavor）](https://github.com/dfinity/internet-identity/blob/main/README.md#build-features-and-flavors) と [agent-js](https://github.com/dfinity/agent-js) ライブラリを使用します。
```

hoosan commented on docs/developer-docs/integrations/internet-identity/integrate-identity.md:
````diff
@@ -1,3 +1,113 @@
+$ dfx deploy --no-wallet --argument '(null)'
+```
+
+この時点で、レプリカ（現実的には Internet Computer のローカル版）が稼働し、3つの Canister が配備されたことになります：
````
practical mattersの別訳suggestionです。
デプロイはそのままで良いと思いました
```suggestion
この時点で、レプリカ（実用上の意味としては Internet Computer のローカル版）が稼働し、3つの Canister がデプロイされたことになります：
```

hoosan commented on docs/developer-docs/integrations/internet-identity/integrate-identity.md:
````diff
@@ -1,3 +1,113 @@
+      };
+  };
+  ```
+IC では Principal は、リクエストを実行する人、またはコールをする人（いわゆるコーラー）の識別子です。すべての呼び出しには有効な Principal が必要です。また、匿名呼び出し用の特別な Principal があります。Internet Identity を使用する場合、[self-authenticating principals](../../../references/ic-interface-spec.md#principals) を使用しています。これは、ラップトップに隠れた秘密鍵があり（TouchID、Windows Hello など）、ブラウザが署名して IC への呼び出しを発行する人物であることを証明するために使用する非常に凝った方法です。
````
「凝った方法」が何を指しているのがわかりにくいので、
「これ（self-authenticating principals）はあなたが秘密鍵をラップトップに有していることを示すための非常に凝った方法」
と最初に述べるのはどうでしょうか。

コーラーは違和感ありませんでした

```suggestion
IC では Principal は、リクエストを実行する人、またはコールをする人（いわゆるコーラー）の識別子です。すべての呼び出しには有効な Principal が必要です。また、匿名呼び出し用の特別な Principal があります。Internet Identity を使用する場合、[self-authenticating principals](../../../references/ic-interface-spec.md#principals) を使用することになります。これは、あなたが秘密鍵をラップトップに有している（TouchID、Windows Hello などの裏に隠れています）ことを示すための非常に凝った方法で、この秘密鍵はブラウザが署名に用いたり、あなたが確かに IC への呼び出しを行なっている人物であると証明するために使用されます。
```

hoosan commented on docs/developer-docs/integrations/internet-identity/integrate-identity.md:
````diff
@@ -1,3 +1,113 @@
+  ```
+IC では Principal は、リクエストを実行する人、またはコールをする人（いわゆるコーラー）の識別子です。すべての呼び出しには有効な Principal が必要です。また、匿名呼び出し用の特別な Principal があります。Internet Identity を使用する場合、[self-authenticating principals](../../../references/ic-interface-spec.md#principals) を使用しています。これは、ラップトップに隠れた秘密鍵があり（TouchID、Windows Hello など）、ブラウザが署名して IC への呼び出しを発行する人物であることを証明するために使用する非常に凝った方法です。
+
+IC が実際に `whoami` Canister に通話（リクエスト）を通した場合、それはすべてがチェックアウトされたことを意味し、`whoami` Canister は IC がリクエストに追加する情報、すなわち、あなたの識別子（Principal）をレスポンスするのみです。
````
上の文で、「あなたが確かにICへの呼び出しを行なっている人物であると証明」していると言っていますが、実際のところICがリクエストに「あなたの識別子」を追加しています。whoami canisterはそれ自体が何か作業しているわけではなく、「ICによってリクエストに追加された情報（識別子）」を単に返しているだけだと言っています
```suggestion
IC が実際に `whoami` Canister への呼び出し（リクエスト）を通した場合、それはすべてが確認されたことを意味し、`whoami` Canister は IC がリクエストに追加した情報（つまりあなたの識別子（Principal））を単に返します。
```

hoosan commented on docs/developer-docs/integrations/internet-identity/integrate-identity.md:
```diff
@@ -1,3 +1,113 @@
+
+DFINITY では、Internet Identity でログインするための[使いやすいライブラリ（agent-js）](https://github.com/dfinity/agent-js) を提供しています。
+
+ログインし、Identity を使って Canister コールに使用するために必要な手順です：
```
```suggestion
以下は、ログインとCanister コールに Identity を使用するために必要な手順です：
```

hoosan commented on docs/developer-docs/integrations/internet-identity/integrate-identity.md:
```diff
@@ -1,3 +1,113 @@
+
+では、その Canister を使ってみましょう。詳細を気にしないなら [ヘルパー](#ヘルパー)まで読み飛ばしてください。
+
+これらの Canister と対話するためには（例えばブラウザで Web アプリを表示するためには）、それぞれの Canister ID を把握して、 `https://localhost:8000/?canisterId=<canister ID>` という形式の URL を使う必要があります（ここで `8000` は `dfx` がレプリカへのプロキシコールに用いるポートです。 このポートは通常 `dfx.json` で指定されています）。Canister ID は `dfx コマンド` の出力で確認することができます。また、 `dfx` の "internal" (read: non-documented) ステートで確認することができます。
```
non-documentedはドキュメント化されていないことを示しているものと思われます
```suggestion
これらの Canister と対話するためには（例えばブラウザで Web アプリを表示するためには）、それぞれの Canister ID を把握して、 `https://localhost:8000/?canisterId=<canister ID>` という形式の URL を使う必要があります（ここで `8000` は `dfx` がレプリカへのプロキシコールに用いるポートです。 このポートは通常 `dfx.json` で指定されています）。Canister ID は `dfx コマンド` の出力で確認することができます。また、 `dfx` の "internal" (ドキュメント化されていません) ステートで確認することができます。
```

hoosan commented on docs/developer-docs/integrations/internet-identity/alternative-origins.md:
```diff
@@ -1,3 +1,115 @@
+
+### アセット Canister を設定する
+
+`.well-known`のドット構文（隠しファイル）は通常、ファイルシステムによって “hidden" として扱われるため、ドキュメントをアップロードするためにはアセット Canister を設定することが必要です。アセット Canister を設定するには、`.ic-assets.json` というファイルを新規に作成します。`.ic-assets.json` は Canister の `sources` にリストされているディレクトリの中に置く必要があります。新しいファイル一覧はこのようになります。
```
隠しファイルをhiddenのところにあてるのはどうでしょうか
```suggestion
`.well-known`のドット構文は通常、ファイルシステムによって “隠しファイル" として扱われるため、ドキュメントをアップロードするためにはアセット Canister を設定することが必要です。アセット Canister を設定するには、`.ic-assets.json` というファイルを新規に作成します。`.ic-assets.json` は Canister の `sources` にリストされているディレクトリの中に置く必要があります。新しいファイル一覧はこのようになります。
```

hoosan reviewed (changes requested):
翻訳ありがとうございます。レビューしましたのでご確認ください。

hokosugi committed:
Apply suggestions from code review

レビュー、ありがとうございました。
read: non-documentedの訳はなるほどと思いました。どうりで検索しても何も見つからない訳ですね。それに気付きさえすれば良かったのにと、悔しい思いです。笑

Co-authored-by: hoosan <40290137+hoosan@users.noreply.github.com>
